projects/plugins/jetpack/tests/php/json-api/test-class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Jetpack `sites/%s/settings` endpoint unit tests.
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * Jetpack `sites/%s/settings` endpoint unit tests.
+ */
+class WP_Test_WPCOM_JSON_API_Site_Settings_Endpoint extends WP_UnitTestCase {
+	/**
+	 * Tests the get_woocommerce_address() method. If the setting requests is not set, it should return ''.
+	 *
+	 * @dataProvider woocommerce_setting_key_names
+	 *
+	 * @param string $setting_name The setting lookup key.
+	 */
+	public function test_get_woocommerce_address_no_settings_or_domain_set( $setting_name ) {
+		$mock_class = new Expose_WPCOM_JSON_API_Site_Settings_Endpoint( array( 'stat' => null ) );
```

Comment:
You could avoid this workaround with a [`ReflectionClass`](https://www.php.net/manual/en/class.reflectionclass.php). There are some examples in the Jetpack repo, [like this one](https://github.com/Automattic/jetpack/blob/7de7ab7d4d50b4428fcd2b563bc7dba326b6d803/projects/plugins/jetpack/tests/php/core-api/wpcom-endpoints/test-class-wpcom-rest-api-v2-endpoint-admin-menu.php#L147-L155), that you could work off of. Searching for ReflectionClass should give you a few more.

Reply:
Since we removed the `get_woocommerce_address` method, it no longer needs testing. So, I totally revamped the unit tests. They test the GET and POST request now.